src/Web/View/Render.hs

```diff
@@ -45,23 +46,25 @@ renderText' c u = intercalate "\n" content
  where
   -- T.intercalate "\n" (content <> style css)
   content :: [Text]
-  content = map (unlines . renderContent) . (.contents) $ runView c addCss
+  content = map (unlines . renderContent indent) . (.contents) $ runView c addCss
 
   addCss = do
-    viewInsertContents [styleElement]
+    forM_ styleElement $ viewInsertContents . pure
     u
 
-  css :: [Text]
-  css = renderCSS $ (.css) $ runView c u
+  css :: Maybe (NonEmpty Text)
+  css = nonEmpty $ renderCSS $ (.css) $ runView c u
+
+  styleElement :: Maybe Content
+  styleElement = case css of
+    Just css' -> Just $ Node $ Element "style" (Attributes [] [("type", "text/css")]) [Text $ intercalate "\n" $ toList css']
```

Comment:
We are handling empty here because you want to use `addClass` to add external classes, right? Can we add some tests for this? 

Did `addClass` feel ergonomic for using external classes? Do we need some new utility methods to support it?